deepchem/data/datasets.py

```diff
@@ -1473,8 +1665,33 @@ def from_numpy(X: np.ndarray,
                  w: Optional[np.ndarray] = None,
                  ids: Optional[np.ndarray] = None,
                  tasks: Optional[Sequence] = None,
-                 data_dir: Optional[str] = None) -> "DiskDataset":
-    """Creates a DiskDataset object from specified Numpy arrays."""
+                 data_dir: Optional[str] = None,
+                 legacy_metadata: bool = False) -> "DiskDataset":
+    """Creates a DiskDataset object from specified Numpy arrays.
+
+    Parameters
+    ----------
+    X: np.ndarray
+      Feature array
+    y: Optional[np.ndarray], optional (default None)
+      labels array
+    w: Optional[np.ndarray], optional (default None)
+      weights array
+    ids: Optional[np.ndarray], optional (default None)
+      identifiers array
+    tasks: Optional[Sequence], optional (default None)
+      Tasks in this dataset
+    data_dir: Optional[str], optional (default None)
+      The directory to write this dataset to. If none is specified, will use
+      a temporary dataset instead.
```

Comment:
I think you meant "a temporary directory instead"?

Reply:
Good catch! Will fix